plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3630,6 +3634,37 @@ public void testEmptyDelete()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCtasFromAVersionedTable()
+            throws Exception
+    {
+        String sourceTableName = "test_ctas_versioned_source_" + randomTableSuffix();
+        String snapshotVersionedSinkTableName = "test_ctas_snapshot_versioned_sink_" + randomTableSuffix();
+        String timestampVersionedSinkTableName = "test_ctas_timestamp_versioned_sink_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + sourceTableName + "(an_integer integer)");
+        Thread.sleep(1);
```

Comment:
What is the point of this Thread.sleep? Next ones make sense but this one I don't understand.

Reply:
The reasoning behind this `Thread.sleep(1)` is to avoid having two snapshots at the timestamp `afterInsert123EpochMillis`

Reply:
add a code comment.